lab_3_ann_retriever/main.py

```diff
@@ -61,6 +76,15 @@ def save_vector(vector: Vector) -> dict:
     Returns:
         dict: A state of the vector to save
     """
+    state: dict[str, int | dict] = {
+        "len": len(vector),
+        "elements": {}
+    }
+    for idx, element in enumerate(vector):
+        if element != 0:
+            if isinstance(state["elements"], dict):
```

Comment:
Do you really need this check?

Reply:
There were problems with MyPy without it :(
How can I tell MyPy that `state["elements"]` has actually dict type?

Reply:
fixed